### PR TITLE
Improvements to stdlib/3/_curses.pyi - take 3

### PR DESCRIPTION
**Pull request by gossrock**
Changed Optional typed arguments to optional arguments to fix to fix issue #1758.
Also updated format to conform with CONTRIBUTING.md conventions.
And added a few missing return types and corrected a few other type annotations.

### CODE REVIEW

File: stdlib/3/_curses.pyi [hunk 1]
```diff
@@ -292,227 +292,171 @@ def use_env(flag: bool) -> None: ...
...
-    def border(self, ls: Optional[int], rs: Optional[int],
-               ts: Optional[int], bs: Optional[int],
-               tl: Optional[int], tr: Optional[int],
-               bl: Optional[int], br: Optional[int]) -> None: ...
-
-    def box(self, vertch: Optional[Tuple[int, int]], horch: Optional[Tuple[int, int]]) -> None: ...
-
+    def bkgd(self, ch: _chtype, attr: int = ...) -> None: ...
+    def bkgset(self, ch: _chtype, attr: int = ...) -> None: ...
+    @overload
```
Comment (JelleZijlstra) [inline stdlib/3/_curses.pyi]: I don't think you need any of these overloads, just the last one is enough. The default of `...` indicates that an argument can be omitted; you only need overloads in special cases like when a function either takes n or n + 2 arguments (like chgat below).

File: stdlib/3/_curses.pyi [hunk 2]
```diff
@@ -292,227 +292,171 @@ def use_env(flag: bool) -> None: ...
...
-
     def cursyncup(self) -> None: ...
-
     @overload
     def delch(self) -> None: ...
-
     @overload
-    def delch(self, y: int, x: int) -> None: ...
-
+    def delch(self, y: int = ..., x: int = ...) -> None: ...
```
Comment (JelleZijlstra) [inline stdlib/3/_curses.pyi]: I don't think `= ...` is correct here; if one of these is given both must be given (if I read the docs right).

File: stdlib/3/_curses.pyi [hunk 3]
```diff
@@ -292,227 +292,171 @@ def use_env(flag: bool) -> None: ...
...
-
     @overload
     def derwin(self, begin_y: int, begin_x: int) -> '_CursesWindow': ...
-
     @overload
     def derwin(self, nlines: int, ncols: int, begin_y: int, begin_x: int) -> '_CursesWindow': ...
-
-    def echochar(self, ch: _chtype, attr: Optional[int]) -> None: ...
-
+    def echochar(self, ch: _chtype, attr: Optional[int] = ...) -> None: ...
```
Comment (JelleZijlstra) [inline stdlib/3/_curses.pyi]: Is this really Optional (is None an allowed value for `attr`)?

File: stdlib/3/_curses.pyi [hunk 4]
```diff
@@ -292,227 +292,171 @@ def use_env(flag: bool) -> None: ...
...
-
-    def getstr(self, y: Optional[int], x: Optional[int]) -> None: ...
-
-    def getyx(self) -> Tuple['_CursesWindow', int, int]: ...
-
+    @overload
+    def getstr(self) -> _chtype: ...
+    @overload
+    def getstr(self, n: int) -> _chtype: ...
+    @overload
```
Comment (JelleZijlstra) [inline stdlib/3/_curses.pyi]: The last two overloads are the same.

File: stdlib/3/curses/__init__.pyi [hunk 5]
```diff
@@ -1,9 +1,10 @@
...
+from typing import Callable, Any, Sequence, Mapping
 
 LINES: int
 COLS: int
 
 def initscr() -> _curses._CursesWindow: ...
-def start_color(): ...
-def wrapper(func, *args, **kwds): ...
+def start_color() -> None: ...
+def wrapper(func: Callable, *args: Sequence[Any], **kwds: Mapping[Any, Any]) -> None: ...
```
Comment (JelleZijlstra) [inline stdlib/3/curses/__init__.pyi]: The type for `*` and `**` arguments is the type of individual elements, not the type of the `args/kwds` variable. You probably just want both of them to be Any. Also, can you make the func argument be `Callable[..., Any]` to be explicit?

File: stdlib/3/_curses.pyi (hunk 2 above)
Comment (gossrock) [reply stdlib/3/_curses.pyi]: Yeah, Thanks for the catch.

File: stdlib/3/_curses.pyi (hunk 1 above)
Comment (gossrock) [reply stdlib/3/_curses.pyi]: The reason I did it the way that I did was from the documentation:
> window.border([ls[, rs[, ts[, bs[, tl[, tr[, bl[, br]]]]]]]])
And ...
>Note:
> A 0 value for any parameter will cause the default character to be used for that parameter. Keyword parameters can not be used.

so all arguments preceding an argument that is needed must be present. So actually I guess I need to remove the '= ...'s to be accurate. 

Though now that I'm thinking about it the fact that they are all the same type then the type checker will not know the difference.

I'll reduce it down to 1 with all optional.

File: stdlib/3/_curses.pyi (hunk 3 above)
Comment (gossrock) [reply stdlib/3/_curses.pyi]: Considering the different experiments that I have done your are probably right. I initially made them all ```Optional[int] = ...``` and then realized my mistake and did the experiments to prove that they threw an exception when None was there. I may have missed this one in that process. I'll do the echochar experiment and get back to you.

File: stdlib/3/_curses.pyi (hunk 4 above)
Comment (gossrock) [reply stdlib/3/_curses.pyi]: Oh, must have been distracted. The last one should be with argments x, y and n.

File: stdlib/3/curses/__init__.pyi (hunk 5 above)
Comment (gossrock) [reply stdlib/3/curses/__init__.pyi]: Ah! I kept running myself in circles on that how that should be typed. 
```def wrapper(func: Callable[..., Any], *arg: Any, **kwds: Any) -> None: ...```

File: stdlib/3/_curses.pyi (hunk 3 above)
Comment (gossrock) [reply stdlib/3/_curses.pyi]: Exception thrown with a None in the second argument.

**Comment by gossrock** [comment]
Okay, I believe I have made all the requested changes.

**Pull request merged**